server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ACVRDownload.java

```diff
@@ -72,51 +69,34 @@ public String endpoint(final Request the_request, final Response the_response) {
     try {
       final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
       final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
-      final Set<CastVoteRecord> matches = getMatching();
-      if (matches == null) {
-        result = "Unable to fetch records from database";
-        status = HttpStatus.INTERNAL_SERVER_ERROR_500;
-      } else {
-        Main.GSON.toJson(matches, bw);
-        bw.flush();
-        result =  "";
-      }
-    } catch (final IOException e) {
-      status = HttpStatus.INTERNAL_SERVER_ERROR_500;
-      result = "Unable to stream response";
-    }
-    
-    the_response.status(status);
-    return result;
-  }
-  
-  /**
-   * @return the set of cast vote records that were submitted by auditors, or 
-   * null if the query fails.
-   */
-  private Set<CastVoteRecord> getMatching() {
-    Set<CastVoteRecord> result = null;
-    
-    try {
+      final JsonWriter jw = new JsonWriter(bw);
       Persistence.beginTransaction();
-      final Session s = Persistence.currentSession();
-      final CriteriaBuilder cb = s.getCriteriaBuilder();
-      final CriteriaQuery<CastVoteRecord> cq = 
-          cb.createQuery(CastVoteRecord.class);
-      final Root<CastVoteRecord> root = cq.from(CastVoteRecord.class);
-      cq.select(root).where(cb.equal(root.get("my_record_type"), 
-                                     RecordType.AUDITOR_ENTERED));
-      final TypedQuery<CastVoteRecord> query = s.createQuery(cq);
-      result = new HashSet<CastVoteRecord>(query.getResultList());
+      jw.beginArray();
+      final Stream<CastVoteRecord> matches = 
+          CastVoteRecordQueries.getMatching(RecordType.AUDITOR_ENTERED);
+      matches.forEach((the_cvr) -> {
+        try {
+          jw.jsonValue(Main.GSON.toJson(the_cvr));
+          Persistence.currentSession().evict(the_cvr);
+        } catch (final IOException e) {
```

Comment:
Actually, we can't, because it's in a lambda. I think the only reason I had to catch the exception at all (instead of letting the `IOException` handler below deal with it) is because of that fact. Maybe I should save it and rethrow it after the `forEach()`...